src/clj/rems/workflow/dynamic.clj

```diff
@@ -636,12 +639,17 @@
   [cmd application _injections]
   (or (actor-is-not-commenter-error application cmd)
       (state-error application ::submitted)
-      {:success true
-       :result {:event/type :application.event/commented
-                :event/time (:time cmd)
-                :event/actor (:actor cmd)
-                :application/id (:application-id cmd)
-                :application/comment (:comment cmd)}}))
+      (let [last-request-for-actor
+            (last (filter #(contains? (set (:application/commenters %))
+                                      (:actor cmd))
+                           (:dynamic-events application)))]
```

Comment:
Don't iterate the events here, but assoc the necessary state to application in the `apply-event [:application.event/comment-requested :workflow/dynamic]` method. (Because the application here has mixed responsibilites - it's both the read model and the write model instead of following CQRS - you'll probably need to dissoc the state before returning the application from the public API. See usages of  `rems.permissions/cleanup` to find the right places.)

Reply:
Also it seems weird to dig the last request ID from the events when it could come as part of the command. Or would that complicate the client unnecessarily?

Reply:
If the comments are always related to the latest comment request, then the use of a request ID is unnecessary because the information can be derived from the existing events.